### ISSUE TITLE
Support for Ubuntu Linux (also Debian)

### ISSUE THREAD

**Issue by jeb2239** [Morgan-Stanley/hobbes#4]
I have had luck building this on ubuntu 17.04 with no issues as long as I use `find_package(LLVM 3.7.1 EXACT REQUIRED CONFIG)` after installing llvm-3.7-dev package from the universes repository. llvm 3.9 and 4.0 will not work

edit:

Actually whenever I try to run hi I get a unification error. I am looking into this.
```
./hi
stdin:60,1-62,22: stdin:61,10-19: Cannot unify types: <std.__cxx11.basic_string<char, std.char_traits<char>, std.allocator<char> >> != <std.string>

```

**Comment by kthielen** [comment]
Hi Jonathan!

I also reproduced this unification error during my effort to port to the Mac and I have a fix -- I had to introduce a special-case lift for std::string (in lib/hobbes/lang/tylift.H) to make sure that it got the expected name (ie: the name "std::string" lifted by default on RHEL5/6 with gcc4.8/5.2).

**Comment by kthielen** [comment]
FYI, this change to tylift.H along with my OSX port pull request should fix the unification error that you're hitting there:

https://github.com/Morgan-Stanley/hobbes/pull/5/commits/a3e58349ca531e92362676a8505211bbfa4dbe41#diff-77655af29ede8b12ac556236fd468abe

**Comment by kthielen** [comment]
Here's a PR that specifically targets these issues with Debian and LLVM 3.7:
https://github.com/Morgan-Stanley/hobbes/pull/6

I installed VirtualBox with the latest Debian OS on my laptop, got everything working with LLVM 3.7.

Sorry for the wait.  I have been working on the OSX port and although I have it working with LLVM 3.5 there, there are some problems building for different versions of LLVM.  Rather than block all progress while OSX is sorted out, I've made this much smaller PR to at least address the issues with Debian and probably many other Linux setups outside of the Morgan Stanley one.
